### PR TITLE
add workflow to execute unit tests

### PR DESCRIPTION
**Pull request by Snoup97**

### CODE REVIEW

File: .github/workflows/pytest-action.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.9
```
Comment (AntonOsika) [inline .github/workflows/pytest-action.yml:19]: We want at least 3.10

**Comment by AntonOsika** [comment]
Can you explain what else is needed to make these tests run in the PR?

**Comment by Snoup97** [comment]
@AntonOsika to enable workflows to run on PRs you need to head to the settings of this repo, to the actions/general section. there should be a section that says "Fork pull request workflows", here you just set the checkbox for the "Run workflows from fork pull requests". the other checkboxes can be left unchecked.

File: .github/workflows/pytest-action.yml (hunk 1 above)
Comment (Snoup97) [reply .github/workflows/pytest-action.yml:19]: alright ill change it

**Comment by Snoup97** [comment]
@AntonOsika ill also change it to run on pytest and move the tests i wrote in #128 to pytest.

**Comment by patillacode** [comment]
If we are going to use pytest, we should add it as a requirement

**Comment by patillacode** [comment]
Moving this into draft until `pytest` is added as a requirement, please mark as ready when done and we can approve and merge!

File: .github/workflows/pytest-action.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.10
```
Comment (patillacode) [inline .github/workflows/pytest-action.yml:19]: I think it need to be something like 3.10.11

build is throwing the error:
```
  Error: The version '3.1' with architecture 'x64' was not found for Ubuntu 22.04.
  The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
```

Comment (Snoup97) [reply .github/workflows/pytest-action.yml]: yep. fixed it

**Comment by carlthome** [comment]
There already is a CI workflow on main for automated unit testing in PRs with pytest.

**Pull request merged**